lorax/lorax.py

```diff
@@ -33,24 +34,340 @@ class TheLorax(object):
+                             'Use TheLorax.set_name_patterns() first.')
+        elif how not in ['features', 'patterns']:
+            # NOTE-KA: Minor, in this case, should we default to features and let the code run with a warning?
+            raise ValueError('How must be one of features or patterns.')
+
+        # TODO: Add error handling for sample's features and the data features.
+        if isinstance(sample, pd.Series):
+            sample = sample.values
+
+        if self.X_test is not None and idx is not None:
+            sample = self.X_test.loc[idx]
+
+            # When creating X_test in load data, 
+            # if the classifier contains a Intercept, it is added as a feature. 
+            # Have to remove it before the inference phase
+            if 'Intercept' in sample.index.values:
+                sample = sample.drop(['Intercept'])
+            
+            sample = sample.values
+
+        # Formatting the test data matrix by setting appropriate index and removing non-feature coulmns
+        if test_mat is not None:            
+            # Indexing and exclusing non-feature columns
+            # NOTE-KA: I think this should be handled outside of Lorax
+            if self.id_col is not None:
+                test_mat.set_index(self.id_col, inplace=True)
+
+            for dr_col in self.drop_cols:
+                # Dropping the non-feature columns in the new test matrix, if they exist
+                # TODO: Handle the ID cols, Date cols elegantly
+                test_mat = test_mat.drop(dr_col, axis=1, errors='ignore')
+
+            if idx is not None:
+                sample = test_mat.loc[idx].values
+        
+        # Calculating Feature contributions
+        if isinstance(self.clf, RandomForestClassifier):
+            # Getting values for Random Forest Classifier
+            return_tuple = get_contrib_list_RF(self.clf, sample, self.column_names)
+
+            # Feature importance scores
+            contrib_list = return_tuple[4]
+        
+            # Model component information
+            self.model_info['num_trees'] = return_tuple[0]
+            self.model_info['global_score_dict'] = return_tuple[1]
+            self.model_info['feature_dict'] = return_tuple[2]
+            self.model_info['aggregated_dict'] = return_tuple[3]        
+            
+        elif isinstance(self.clf, LogisticRegression):
+            # Getting values for Random Forest Classifier
+            # TODO: The column names need to be consolidated
+
+            contrib_list = get_contrib_list_LR(self.clf, sample, self.column_names)
+
+        # Setting the prediction class
+        # predict proba returns a 2D array
+        scores = self.clf.predict_proba(sample.reshape(1, -1))[0]  
+
+        if pred_class is None:
+            # TODO: Multiclass adpatation
+            # use np.argmax(), or clf.predict()
+            pred_class = np.argmax(scores)
```

Comment:
I would probably log a warning here since in most real cases we would likely have some other threshold/top k, so if we get here it's probably more likely a mistake than intentional.